base/common/src/org/dogtagpki/nss/NSSDatabase.java

```diff
@@ -343,13 +348,61 @@ public void addPEMCertificate(
         }
     }
 
+    public void addBasicConstraintsExtension(
+            List<String> cmd,
+            PrintWriter out,
+            BasicConstraintsExtension extension) throws Exception {
+
+        logger.info("Adding basic constraints extension:");
+
+        cmd.add("-2");
+
+        // Is this a CA certificate [y/N]?
+        boolean ca = (boolean) extension.get(BasicConstraintsExtension.IS_CA);
+        logger.info("- CA: " + ca);
+        if (ca) {
+            out.print("y");
+        }
+        out.println();
+
+        // Enter the path length constraint, enter to skip [<0 for unlimited path]: >
```

Comment:
Question: Doesn't `path_len=0` mean that the CA can have no intermediate entities and `-1` or `none` mean unlimited? The comment is slightly misleading

Reply:
That's correct, but the comment above is coming from `certutil`. The code here is generating the input for `certutil` for that particular question.

Reply:
It is `<0` for unlimited path length. So, `-1`.

Reply:
oh oh! My bad! I read the whole text between `<>`. I missed the `:`. Ugh! Marking this as resolved